openff/interchange/common/_nonbonded.py

```diff
@@ -112,7 +111,7 @@ def charges(
     ) -> dict[TopologyKey | LibraryChargeTopologyKey | VirtualSiteKey, _ElementaryChargeQuantity]:
         """Get the total partial charge on each atom, including virtual sites."""
         if len(self._charges) == 0 or self._charges_cached is False:  # type: ignore[has-type]
-            self._charges = self._get_charges(include_virtual_sites=False)
+            self._charges.update(self._get_charges(include_virtual_sites=False))
```

Comment:
This ... makes a fair amount of sense to me, actually. I think in non-Pydantic world it's best practice to update a dict instead of re-assign it, but I'm also pretty sure something in Pydantic's magic machinery means just passing the return value of another function to an instance attribute gummed something up something.